contracts/test-utils/src/unit_test_helper.ts

```diff
@@ -0,0 +1,82 @@
+import { RevertError } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+
+import { ContractArtifact, TransactionReceiptWithDecodedLogs } from 'ethereum-types';
+
+import { expect } from './chai_setup';
+import { LogDecoder } from './log_decoder';
+
+type AsyncFunction<TArgs extends any[], TResult> = (...args: TArgs) => Promise<TResult>;
+
+interface ContractArtifacts {
+    [contractName: string]: ContractArtifact;
+}
+
+interface TransactionReturnData<TCallAsyncResult> {
+    result: TCallAsyncResult;
+    receipt: TransactionReceiptWithDecodedLogs;
+}
+
+interface ContractWrapperFunction<
+    TCallAsyncArgs extends any[],
+    TAwaitTransactionSuccessAsyncArgs extends any[],
+    TCallAsyncResult
+> {
+    callAsync: AsyncFunction<TCallAsyncArgs, TCallAsyncResult>;
+    sendTransactionAsync?: AsyncFunction<TAwaitTransactionSuccessAsyncArgs, string>;
+}
+
+export type TransactionExpectation<TAwaitTransactionSuccessAsyncArgs extends any[], TCallAsyncResult> = (
+    tx: Partial<TransactionReturnData<TCallAsyncResult>>,
+    // tslint:disable-next-line: trailing-comma
+    ...args: TAwaitTransactionSuccessAsyncArgs
+) => Promise<void> | void;
+
+export class BaseUnitTestHelper<TContract> {
+    protected readonly _testContract: TContract;
+    private readonly _logDecoder: LogDecoder;
```

Comment:
Do we actually need `LogDecoder`? Our contract wrappers should be able to handle this, provided that the write `LogEventArg` type is passed in for casting. We could maybe avoid using this by accepting a template argument that specifies the log type that the function will be used for.